docs/api/graphql/querying-entities.md

```diff
@@ -0,0 +1,205 @@
+# Querying Metadata Entities 
+
+### Learn how to fetch & update Entities in your Metadata Graph programmatically
+
+## Queries: Reading an Entity 
+
+DataHub provides the following GraphQL queries for retrieving entities in your Metadata Graph. 
+
+### Getting a Metadata Entity 
+
+To retrieve a Metadata Entity by primary key (urn), simply use the `<entityName>(urn: String!)` GraphQL Query. 
+
+For example, to retrieve a User entity, you can issue the following GraphQL Query:
+
+*As GraphQL*
+
+```graphql 
+{
+  corpUser(urn: "urn:li:corpuser:datahub") {
+    username
+    urn
+  }
+}
+```
+
+*As CURL*
+
+```curl
+curl --location --request POST 'http://localhost:8080/api/graphql' \
+--header 'X-DataHub-Actor: urn:li:corpuser:datahub' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+   "query":"{
+   \n  corpUser(urn: \"urn:li:corpuser:datahub\") {
+   \n    username
+   \n    urn
+   \n  }
+   \n}",
+   "variables":{}}'
+```
+
+### Searching for a Metadata Entity 
+
+To perform full-text search against an Entity of a particular type, use the `search(input: SearchInput!)` GraphQL Query.
+
+As GraphQL:
+
+```graphql 
+{
+  search(input: { type: "DATASET", query: "my sql dataset", start: 0, count: 10 }) {
+    start
+    count
+    total
+    searchResults {
+      entity {
```

Comment:
maybe add the entity agnostic `urn` and `type` fields above `...on Dataset` so the user will see data even if they change entity type?

Reply:
good idea!